### PR TITLE
SAK-46889 Rubric | Export rubric with period followed by more characters leads to bad export

### PR DESCRIPTION
**Pull request by stetsche**
Make filename of rubrics pdf export safer for
browsers and other programs.

### CODE REVIEW

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java [hunk 1]
```diff
@@ -370,7 +371,9 @@ public ResponseEntity<byte[]> getPdf(@PathVariable String siteId, @PathVariable
         checkSakaiSession();
 
         ContentDisposition contentDisposition = rubricsService.getRubric(rubricId).map(rubric -> {
-            return ContentDisposition.builder("attachment").filename(rubric.title + ".pdf").build();
+            String filename = StringUtils.trimToEmpty(rubric.title).replace(".", "_");
+            filename = StringUtils.isNotBlank(filename) ? filename : "_";
+            return ContentDisposition.builder("attachment").filename(String.format("%s.pdf", filename)).build();
```
Comment (adrianfish) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java:376]: I, personally, think using format on such short concatenations doesn't really help with readability. The compiler optimises these concatenations anyway, it replaces them with StringBuilder calls. Just my opinion though, nothing more :)

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java [hunk 2]
```diff
@@ -370,7 +371,9 @@ public ResponseEntity<byte[]> getPdf(@PathVariable String siteId, @PathVariable
         checkSakaiSession();
 
         ContentDisposition contentDisposition = rubricsService.getRubric(rubricId).map(rubric -> {
-            return ContentDisposition.builder("attachment").filename(rubric.title + ".pdf").build();
+            String filename = StringUtils.trimToEmpty(rubric.title).replace(".", "_");
```
Comment (adrianfish) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java:374]: This is good, but we should maybe block using periods when creating the rubric, also. I'm not suggesting do it here, just something to add in the future.

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java [hunk 3]
```diff
@@ -370,7 +371,9 @@ public ResponseEntity<byte[]> getPdf(@PathVariable String siteId, @PathVariable
...
+            filename = StringUtils.isNotBlank(filename) ? filename : "_";
```
Comment (adrianfish) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java:375]: I think this should throw an exception of some kind, or at least log something. A blank rubric title is not good - if blank rubric titles are possible, we should prevent that ever happening in the first place.

**Review by adrianfish — CHANGES_REQUESTED** [review]
The only concern is the possibility of a blank rubric title, resulting in a blank pdf file. We need to prevent that happening and throw an error for existing cases.

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java (hunk 1 above)
Comment (stetsche) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java:376]: Okay, I'll keep it in mind. For next time ;)

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java (hunk 2 above)
Comment (stetsche) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java:374]: Maybe. It's an edge case for sure, but: Can it break something? Is there no value in it?

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java (hunk 3 above)
Comment (stetsche) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java:375]: I think we are on the same page, but just to be clear. This is not in the scope of the issue, right? With this PR I am trying to avoiding creating problematic filenames.

Comment (msoupcoff) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java:375]: Correct; on same page as separate but semi-related issues. I created https://sakaiproject.atlassian.net/browse/SAK-47400 to specifically address the blank name validation. Mentioned the weighted grading validation check as an example. In those instances, the **Save** button gets disabled until the issue is corrected so ideally that same behavior could occur with the title field.

FYI rubrics do get a default name of 'New rubric'. IMO the title field should start as blank and force an instructor to enter a name themself. Seen way too many instances of where an instructor doesnt notice that field and thus saves almost all their rubrics as 'New rubric'. Becomes more complicated when instructor then connects a rubric to an assignment and that locks in that rubric title.

**Review by mpellicer — APPROVED** [review]

**Comment by mpellicer** [comment]
Well done Markus

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java (hunk 3 above)
Comment (stetsche) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java:375]: Thanks, perfect! :)
The described behavior for creating a rubric sounds good to me :+1:

**Pull request merged**